xapian-applications/omega/templates/query

```diff
@@ -153,3 +153,4 @@ $map{$cgilist{B},<input type=hidden name=B value="$html{$_}">}
 </body>
 </html>
 $log{query.log}
+$log{search.log,$hash{md5,$transform{s/^\s+|\s+$$//g,\1,$query}}$.$csv{$query,1}$.$csv{$hitlist{$id$if{$ne{$hit,$sub{$last,1}},$.}},1}$.$topdoc}
```

Comment:
`$cgi{QUERYID}` will only be set in the `clicklog` template (and it is used there).

The issue I'm trying to highlight is with the `query` template - in two different places in this template, the query is munged with a regexp and then an MD5 hash of the result is calculated:

```
$ grep -n -F --colour '$hash{md5,$transform{s/^\s+|\s+$$//g,\1,$query}}' templates/query 
109:<td><b><a href="$if{$opt{clicklog},$html{$or{$env{SCRIPT_NAME},omega.cgi}?FMT=$url{clicklog}&QUERYID=$url{$hash{md5,$transform{s/^\s+|\s+$$//g,\1,$query}}}&DOCID=$url{$id}&URL=$url{$field{url}}},$html{$field{url}}}">$html{$or{$field{caption},$field{title},$prettyurl{$field{url}},Untitled}}</a></b><br>
156:$if{$opt{clicklog},$log{search.log,$hash{md5,$transform{s/^\s+|\s+$$//g,\1,$query}}$.$csv{$query,1}$.$csv{$hitlist{$id$if{$ne{$hit,$sub{$last,1}},$.}},1}$.$topdoc}}
```

If you run it in the terminal yourself you'll get the colour highlighting which makes it clearer.

It'd be better to just calculate this queryid once - the usual approach for this sort of thing is to store it into an option with `$set{queryid,...}` which you can then read with `$opt{queryid}`.